### PR TITLE
reside-340 Conditionally running Ingest Pipelines with Filebeat, Docker and Elastic

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Blog post on my recent adventures in extracting nginx log information for MINT

### CODE REVIEW

File: content/blog/elastic-conditional-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+Index patterns allow post-ingestion processing of data to generate new fields which could be used in further reporting.
+So it is possible that we could have done further processing of the `nginx.clientip` IP addresses in an index pattern. We wanted to use 
+the `GeoIP` processor to map IP addresses to geographical data, which should have been possible using the "Painless" (!) 
+scripting language's ability to invoke processors 
+(described [here](https://www.elastic.co/guide/en/elasticsearch/painless/master/painless-ingest.html#painless-ingest)),
+again combined with conditional application.
+
+However, using index patterns is expensive at query time, and Elastic limits what fields can be used in index patterns.
+Furthermore, we really didn't want to have to recreate the existing `nginx` pipeline with some combination of configured
+and scripted processors - we just wanted to use the pipeline for some documents and not others,
```
Comment (r-ash) [inline content/blog/elastic-conditional-pipelines.md:132]:
```suggestion
and scripted processors - we just wanted to use the pipeline for some documents and not others.
```

Comment (r-ash) [inline content/blog/elastic-conditional-pipelines.md]:
```suggestion
and scripted processors - we just wanted to use the pipeline for some documents and not others.
```

**Review by r-ash — APPROVED** [review]
This looks great, it does make me wonder if we should be adding logstash to our elastic & kibana deployments. We're adding more and more transformations into the elastic & filebeat configurations when really I think it's more typical to do that work in logstash. It would keep all of this configuration in one place and hopefully make it obvious in future where to add more transformations.

Maybe worth us having a review of that at some point. I can't recall why we didn't deploy logstash as part of the stack from the start?

**Review by richfitz — APPROVED** [review]

**Review by M-Kusumgar — APPROVED** [review]
As discussed, I thought it was a pleasant read. Having little to no context of many of the tools used, I found it quite easy to follow. Seems like there's a lot of potential of having conditional pipelines!

**Pull request merged**